staging/src/k8s.io/legacy-cloud-providers/vsphere/vsphere.go

```diff
@@ -894,7 +901,16 @@ func (vs *VSphere) LoadBalancer() (cloudprovider.LoadBalancer, bool) {
 }
 
 func (vs *VSphere) isZoneEnabled() bool {
-	return vs.cfg != nil && vs.cfg.Labels.Zone != "" && vs.cfg.Labels.Region != ""
+	isEnabled := vs.cfg != nil && vs.cfg.Labels.Zone != "" && vs.cfg.Labels.Region != ""
+	// Return false within kubelet in case of credentials stored in secret.
+	// Otherwise kubelet will not be able to obtain zone labels from vSphere and create initial node
+	// due to no credentials at this step.
+	// See https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet_node_status.go#L394 for more details.
```

Comment:
Could you pin this to a commit, since master branch could be changed. I think you are referring to this part https://github.com/kubernetes/kubernetes/blob/bad4faf1b9d69d3cb6e8de90dc5eee1d8cf0b6d3/pkg/kubelet/kubelet_node_status.go#L384-L386